src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,425 @@
+    def __init__(self, *args, **kwargs):
+        """Initialize a IorTestBase object."""
+        super(ServerFillUp, self).__init__(*args, **kwargs)
+        self.no_of_pools = 1
+        self.capacity = 1
+        self.no_of_servers = 1
+        self.no_of_drives = 1
+        self.pool = None
+        self.dmg = None
+        self.container_info = {}
+        self.set_faulty_device = False
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+        # Start the servers and agents
+        super(ServerFillUp, self).setUp()
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        #Get this IOR Parameter from yaml file
+        self.ior_flags = self.params.get("ior_flags",
+                                         '/run/ior/iorflags/*')
+        self.ior_read_flags = self.params.get("ior_flags",
+                                              '/run/ior/iorflags/*', '-r -R')
+        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.ior_transfer_size = str(self.params.get("transfer_size",
+                                                     '/run/ior/iorflags/*'))
+        self.ior_daos_oclass = self.params.get("obj_class",
+                                               '/run/ior/iorflags/*', "SX")
+        self.processes = self.params.get("slots", "/run/ior/clientslots/*")
+        #Get the number of daos_io_servers
+        self.daos_io_servers = (self.server_managers[0].manager
+                                .job.yaml.server_params)
+        self.out_queue = queue.Queue()
+
+    def get_max_capacity(self, drive_info):
+        """Get NVMe storage capacity based on NVMe disk from server yaml file.
+
+        Args:
+            drive_info(list): List of disks from each daos_io_servers
+
+        Returns:
+            int: Maximum NVMe storage capacity.
+
+        """
+        # Get the Maximum storage space among all the servers.
+        drive_capa = []
+        for server in self.hostlist_servers:
+            for daos_io_server in range(len(self.daos_io_servers)):
+                drive_capa.append(sum(drive_info[server][daos_io_server]))
+        print('Maximum Storage space from the servers is {}'
+              .format(int(min(drive_capa) * 0.99)))
+
+        #Return the 99% of storage space as it wont be used 100% for
+        #pool creation.
+        return int(min(drive_capa) * 0.99)
+
+    def get_nvme_lsblk(self):
+        """Get NVMe lsblk from servers.
+
+        Args:
+            None
+
+        Returns:
+            dict: Dictionary of server mapping with disk ID and size
+                  'wolf-A': {'nvme2n1': '1600321314816'}.
+        """
+        nvme_data = {}
+
+        task = run_task(self.hostlist_servers, "lsblk -b /dev/nvme*n*")
+        for _rc_code, _node in task.iter_retcodes():
```

Comment:
Any idea why we need to use lsblk command instead of dmg commands? DMG commands should be able to provide the NVME information. In this way, we will test the dmg command functionality.

Reply:
Right now DMG command does not provide the drives used by server instead it provides all the drives attached to the system. For example if we have 8 drives and user tried to use 2 drive in server.yaml it still display the all attached drives which is not correct. Currently DAOS-4620 open to address it.